descope/management/outbound_application.py

```diff
@@ -0,0 +1,435 @@
+        token_url: Optional[str] = None,
+        token_url_params: Optional[List[URLParam]] = None,
+        revocation_url: Optional[str] = None,
+        default_scopes: Optional[List[str]] = None,
+        default_redirect_url: Optional[str] = None,
+        callback_domain: Optional[str] = None,
+        pkce: Optional[bool] = None,
+        access_type: Optional[AccessType] = None,
+        prompt: Optional[List[PromptType]] = None,
+    ) -> dict:
+        """
+        Create a new outbound application with the given name. Outbound application IDs are provisioned automatically, but can be provided
+        explicitly if needed. Both the name and ID must be unique per project.
+
+        Args:
+        name (str): The outbound application's name.
+        description (str): Optional outbound application description.
+        logo (str): Optional outbound application logo.
+        id (str): Optional outbound application ID.
+        client_secret (str): Optional client secret for the application.
+        client_id (str): Optional client ID for the application.
+        discovery_url (str): Optional OAuth discovery URL.
+        authorization_url (str): Optional OAuth authorization URL.
+        authorization_url_params (List[URLParam]): Optional authorization URL parameters.
+        token_url (str): Optional OAuth token URL.
+        token_url_params (List[URLParam]): Optional token URL parameters.
+        revocation_url (str): Optional OAuth token revocation URL.
+        default_scopes (List[str]): Optional default OAuth scopes.
+        default_redirect_url (str): Optional default redirect URL.
+        callback_domain (str): Optional callback domain.
+        pkce (bool): Optional PKCE (Proof Key for Code Exchange) support.
+        access_type (AccessType): Optional OAuth access type.
+        prompt (List[PromptType]): Optional OAuth prompt parameters.
+
+        Return value (dict):
+        Return dict in the format
+             {"app": {"id": <id>, "name": <name>, "description": <description>, "logo": <logo>}}
+
+        Raise:
+        AuthException: raised if create operation fails
+        """
+        uri = MgmtV1.outbound_application_create_path
+        response = self._auth.do_post(
+            uri,
+            OutboundApplication._compose_create_update_body(
+                name,
+                description,
+                logo,
+                id,
+                client_secret,
+                client_id,
+                discovery_url,
+                authorization_url,
+                authorization_url_params,
+                token_url,
+                token_url_params,
+                revocation_url,
+                default_scopes,
+                default_redirect_url,
+                callback_domain,
+                pkce,
+                access_type,
+                prompt,
+            ),
+            pswd=self._auth.management_key,
```

Comment:
It should work with the user jwt as well, no?

Reply:
yes but not for this function. only the fetch tokens functions should be able to work with that.